src/ResourceManagement/ServerManagement/ServerManagement.Tests/ServerManagementTestBase.cs

```diff
@@ -0,0 +1,271 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Security.Principal;
+using System.ServiceProcess;
+using System.Threading.Tasks;
+using Microsoft.Azure.Management.ServerManagement;
+using Microsoft.Azure.Test.HttpRecorder;
+using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+using Xunit.Abstractions;
+
+
+namespace ServerManagement.Tests
+{
+    public class ServerManagementTestBase : TestBase
+    {
+        protected static string ResourceGroup = "sdktest";
+        private static bool _once;
+
+        private readonly ITestOutputHelper _output;
+
+        protected static bool Recording
+        {
+            get { return HttpMockServer.Mode == HttpRecorderMode.Record; }
+        }
+
+        protected static bool ReuseExistingGateway
+        {
+            get
+            {
+                return Recording &&
+                       "true".Equals(Environment.GetEnvironmentVariable("SMT_REUSE_EXISTING_GATEWAY"),
+                           StringComparison.OrdinalIgnoreCase);
+            }
+        }
+
+        protected static string NodeName
+        {
+            get { return Environment.GetEnvironmentVariable("SMT_NODE_NAME"); }
+        }
+
+        protected static string NodeUserName
+        {
+            get { return Environment.GetEnvironmentVariable("SMT_NODE_USERNAME"); }
+        }
+
+        protected static string NodePassword
+        {
+            get { return Environment.GetEnvironmentVariable("SMT_NODE_PASSWORD"); }
+        }
+
+
+        protected static bool IsAdmin
+        {
+            get
+            {
+                return Extensions.Safe(
+                        () =>
+                            new WindowsPrincipal(WindowsIdentity.GetCurrent()).IsInRole(WindowsBuiltInRole.Administrator));
+            }
+        }
+
+        public ServerManagementTestBase(ITestOutputHelper output)
+        {
+            // add environment variables so that we can just run from VS to record.
+            Extensions.Default("TEST_HTTPMOCK_OUTPUT",
+                String.Format("{0}\\SessionRecords", Directory.GetParent(this.GetType().Assembly.Location).FullName));
+            Extensions.Default("TEST_CSM_ORGID_AUTHENTICATION", "SubscriptionId=3e82a90d-d19e-42f9-bb43-9112945846ef;BaseUri=https://management.azure.com/;AADAuthEndpoint=https://login.windows.net/");
+
+            Extensions.Default("AZURE_TEST_MODE", "Playback");
+            HttpMockServer.Mode = "record".Equals(Environment.GetEnvironmentVariable("AZURE_TEST_MODE"), StringComparison.OrdinalIgnoreCase) ? HttpRecorderMode.Record : HttpRecorderMode.Playback;
```

Comment:
`HttpMockServer.Mode` doesn't get set until after I'd like to know what state we're in
I preemptively set it to what it will get set to later anyway.

I've added a comment to explain this.